src/transaction/TransactionController.ts

```diff
@@ -1212,18 +1238,175 @@ export class TransactionController extends BaseController<
   }
 
   /**
-   * Fucntion to determine if the transaction is in a final state
+   * Resolves the locally stored transactions with the blockchain or etherscan to update TransactionController State
+   * @param remoteTxs - Array of transactions fetched from etherscan, the blockchain or other source
+   * @param localTxs - Array of transactions currently stored in the state of the controller
+   * @param stateReconcileMethod - Strategy used to reconcile the transactions
+   * @returns [boolean, TransactionMeta[]]
+   */
+  private transactionStateReconciler(
+    remoteTxs: TransactionMeta[],
+    localTxs: TransactionMeta[],
+    stateReconcileMethod: StateReconcileMethod,
+  ): [boolean, TransactionMeta[]] {
+    switch (stateReconcileMethod) {
+      case StateReconcileMethod.ETHERSCAN:
+        return this.etherscanTransactionStateReconciler(remoteTxs, localTxs);
+      default:
+        return [false, []];
+    }
+  }
+
+  /**
+   * Method to determine if the transaction is in a final state
    * @param status - Transaction status
    * @returns boolean if the transaction is in a final state
    */
-  private isFinalState(status: TransactionStatus) {
+  private isFinalState(status: TransactionStatus): boolean {
     return (
       status === TransactionStatus.rejected ||
       status === TransactionStatus.confirmed ||
       status === TransactionStatus.failed ||
       status === TransactionStatus.cancelled
     );
   }
+
+  private etherscanTransactionStateReconciler(
+    remoteTxs: TransactionMeta[],
+    localTxs: TransactionMeta[],
+  ): [boolean, TransactionMeta[]] {
+    const outdatedTxs: TransactionMeta[] = this.getOutdatedTransactions(
```

Comment:
nit: This variable name made me think these transactions were outdated, when in fact these are the _updated_ versions of local transactions.

Reply:
Thats a good point, I'll change the name